src/grid_map.rs

```diff
@@ -0,0 +1,27 @@
+// grid map definition
+// author:Salah Eddine Ghamri (s.ghamri)
+
+use std::ops::Deref;
+extern crate nalgebra as na;
+
+pub struct Map{
+    _grid: na::DMatrix<i32>
```

Comment:
According to Rust conventions, please do not use an underscore prefix for field names in a struct. It is preferable to name it grid here.